src/Skia/Perspex.Skia/FormattedTextImpl.cs

```diff
@@ -75,36 +108,206 @@ public Size Measure()
 
         public void SetForegroundBrush(IBrush brush, int startIndex, int length)
         {
+            // Is this method even relevant now???
```

Comment:
This is definitely required. Your formatted text implementation needs to track which ranges different colours are set to, then in the DrawString method use that to split into several draws on different colours.

In the other backends things are simpler because that has this functionality inbuilt.